File: app/src/main/java/com/koktajlbar/incognitobook/utils/UUIDTypeConverter.java
package com.koktajlbar.incognitobook.utils;


import androidx.room.TypeConverter;

import org.jetbrains.annotations.Contract;

import java.util.UUID;

public class UUIDTypeConverter
{
    @Contract("null -> null")
    @TypeConverter
    public static UUID toUUID(String value)
    {
        return value == null ? null : UUID.fromString(value);
    }
    @Contract("null -> null")
    @TypeConverter
    public static String toString(UUID value)
    {
        return value == null ? null : value.toString();
    }
}

File: app/src/main/java/com/koktajlbar/incognitobook/utils/UUIDTypeConverter.kt
package com.koktajlbar.incognitobook.utils

import androidx.room.TypeConverter
import org.jetbrains.annotations.Contract
import java.util.*

object UUIDTypeConverter {
    @Contract("null -> null")
    @TypeConverter
    @JvmStatic
    fun toUUID(value: String?): UUID? {
        return if (value == null) null else UUID.fromString(value)
    }

    @Contract("null -> null")
    @TypeConverter
    @JvmStatic
    fun toString(value: UUID?): String? {
        return value?.toString()
    }
}